CodeMaid/UI/Dialogs/Options/Cleaning/CleaningUpdateViewModel.cs

```diff
@@ -227,5 +228,41 @@ public bool UpdateSingleLineMethods
         }
 
         #endregion Options
+
+        #region AddHeaderVariable Command
+
+        private DelegateCommand _addHeaderVariableCommand;
+
+        /// <summary>
+        /// Gets the set dialog result command.
+        /// </summary>
+        public DelegateCommand AddHeaderVariableCommand => _addHeaderVariableCommand ?? (_addHeaderVariableCommand = new DelegateCommand(OnAddVaribleCommandExecuted));
+
+        /// <summary>
+        /// Called when the <see cref="AddHeaderVariableCommand" /> is executed.
+        /// </summary>
+        /// <param name="parameter">The command parameter.</param>
+        private void OnAddVaribleCommandExecuted(object parameter)
+        {
+            var values = (object[])parameter;
+            if (values.Length != 2)
+            {
+                return;
+            }
+
+            var tabControl = values[0];
+            var variableText = values[1] as string;
+            var textBox = (((tabControl as System.Windows.Controls.TabControl)?.SelectedItem as System.Windows.Controls.TabItem)?.Content as System.Windows.Controls.TextBox);
```

Comment:
Having the view model have hard-coded view references within it breaks the intended flow of View->ViewModel.  It looks like this is just trying to inject some text placeholders.  This could be done by passing the text through the CommandParameter and binding that in the UI.  Then the ViewModel wouldn't need to understand TabControl's, SelectedItems, etc.

Reply:
Okay, so I don't totally get how you want to do that. Here is what we need to have
1- textbox => we need to get the currentposition of the cursor to insert the text at the right place
2- text to add => the variable

How do you suggest binding everything together in the view to be able to just pass the text to add as parameter to the command ?

Reply:
What if you passed the text, the index (i.e. cursor position), and the variable name?

Reply:
This doesn't work as well. My idea was to have something like that :
```
<Button Name="Project" Content="$PROJECT$" Width="Auto" Margin="2, 2, 2, 2" Command="{Binding AddHeaderVariableCommand}">
    <Button.CommandParameter>
        <MultiBinding Converter="{x:Static cnv:AppendParametersConverter.Default}">
            <Binding ElementName="LanguageTab" Path="SelectedItem.Content.SelectionStart"/>
            <Binding ElementName="LanguageTab" Path="SelectedItem.Content.Text"/>
            <Binding ElementName="FileName" Path="Content" />
        </MultiBinding>
    </Button.CommandParameter>
</Button>
```

First, this still requires the elementName (mostly because the tabcontrol is not a direct ancestor, but the sibling of an ancestor)
Second we want to bind to the cursor position ( it needs to be updated after we add the text) and bind to the text... which doesn't work with this approach.
Third I tried to find a way to bind in xaml directly to a control of the the selected tab item without  much success...

Reply:
By the way, just realised that the start selection from a textbox is not bindable... lol